go/kbfs/libkbfs/prefetcher.go

```diff
@@ -1299,6 +1304,75 @@ func (p *blockPrefetcher) handlePrefetchRequest(req *prefetchRequest) {
 	}
 }
 
+func (p *blockPrefetcher) handleAppStateChange(
+	appState *keybase1.MobileAppState) {
+	// Pause the prefetcher on cell connections.
+	for *appState != keybase1.MobileAppState_FOREGROUND {
+		p.log.CDebugf(
+			context.TODO(), "Pausing prefetcher while backgrounded")
+		select {
+		case *appState = <-p.appStateUpdater.NextAppStateUpdate(
+			appState):
+		case req := <-p.prefetchStatusCh.Out():
+			p.handleStatusRequest(req.(*prefetchStatusRequest))
+			continue
+		case <-p.almostDoneCh:
```

Comment:
Why `almostDoneCh` and not `shutdownCh`? I guess you could be keeping this handler alive just to wait for `inFlightFetches`, so that the app/net state changes are respected even while we're shutting down. But it seems like if we're shutting down we just want it all to be over with as fast as possible...

Reply:
Yeah I just wanted to make sure that the `run()` loop gets to do all the stuff it does in the `<-almostDoneCh` case.  I didn't want to skip over that stuff -- the timing of everything here is so delicate.

Reply:
Fair enough. I agree the ordering is very brittle.